### PR TITLE
Add search field to search order by user email

### LINKED ISSUE #397: Add search field to search order by user email
image

### PR DESCRIPTION
**Pull request by OliaHrytsak**
image

image

https://github.com/user-attachments/assets/2460a8c6-bcab-45f7-bc96-58482fb5b585

### CODE REVIEW

File: code/src/components/app-search-input/AppSearchInput.test.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      <AppSearchInput
+        placeholder={placeholder}
+        onClear={handleClear}
+        onSearch={handleSearch}
+        value="test"
+      />
+    );
+  });
+  test("Should render search input field", () => {
+    const searchInputField = screen.getByPlaceholderText("Search...");
```
Comment (mhevyk) [inline code/src/components/app-search-input/AppSearchInput.test.tsx]: Reuse existing above `placeholder` variable

File: code/src/components/app-search-input/AppSearchInput.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import ClearIcon from "@mui/icons-material/Clear";
+import SearchIcon from "@mui/icons-material/Search";
+import { InputBase, Paper } from "@mui/material";
+
+import "@/components/app-search-input/AppSearchInput.scss";
+
+import AppIconButton from "../app-icon-button/AppIconButton";
```
Comment (mhevyk) [inline code/src/components/app-search-input/AppSearchInput.tsx]: Use alias here

File: code/src/components/app-search-input/AppSearchInput.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import ClearIcon from "@mui/icons-material/Clear";
+import SearchIcon from "@mui/icons-material/Search";
+import { InputBase, Paper } from "@mui/material";
```
Comment (mhevyk) [inline code/src/components/app-search-input/AppSearchInput.tsx]: Use default import

File: code/src/components/app-search-input/AppSearchInput.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import "@/components/app-search-input/AppSearchInput.scss";
+
+import AppIconButton from "../app-icon-button/AppIconButton";
+import { AppSearchInputProps } from "./AppSearchInput.types";
```
Comment (mhevyk) [inline code/src/components/app-search-input/AppSearchInput.tsx]: Alias

File: code/src/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.constants.ts [hunk 5]
```diff
@@ -1,4 +1,7 @@
-import { AdminOrderFilters } from "@/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.types";
+import {
+  AdminOrderFilters,
+  AdminOrderSearch
+} from "@/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.types";
```
Comment (mhevyk) [inline code/src/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.constants.ts:4]: Fix formatting here

File: code/src/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.ts [hunk 6]
```diff
@@ -1,6 +1,9 @@
 import { useSearchParams } from "react-router-dom";
 
-import { defaultAdminOrderFilters } from "@/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.constants";
+import {
+  defaultAdminOrderFilters,
+  defaultSearchOrderFilter
+} from "@/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.constants";
```
Comment (mhevyk) [inline code/src/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.ts:6]: Same with formatting

File: code/src/pages/dashboard/dashboard-orders/DashboardOrdersPage.tsx [hunk 7]
```diff
@@ -21,14 +23,39 @@ const DashboardOrdersPage = () => {
...
     totalPages,
     isLoading
   } = useFilteredAdminOrders();
 
+  const { applyFilters, resetFilterByKey, updateFilterByKey } = searchActions;
+
   const [isFilterDrawerOpen, setIsFilterDrawerOpen] = useState(false);
+  const [searchValue, setSearchValue] = useState(
+    searchFilters.accountEmail || ""
+  );
```
Comment (mhevyk) [inline code/src/pages/dashboard/dashboard-orders/DashboardOrdersPage.tsx]: We don't need this state. Use `searchFilters.accountEmail` as value and `updateFilterByKey` as updator function

**Review by mhevyk — COMMENTED** [review]
Good work!

File: code/src/pages/dashboard/dashboard-orders/messages/uk.json [hunk 8]
```diff
@@ -1,3 +1,4 @@
 {
-  "dashboardTabs.orders.title": "Замовлення"
+  "dashboardTabs.orders.title": "Замовлення",
+  "dashboardTabs.orders.search": "Шукати за ел.адресою..."
```
Comment (heorhiiss) [inline code/src/pages/dashboard/dashboard-orders/messages/uk.json]: I guess there should be space between `ел.` and `адресою`

File: code/src/pages/dashboard/dashboard-orders/DashboardOrdersPage.scss [hunk 9]
```diff
@@ -3,6 +3,11 @@
   align-items: center;
   justify-content: space-between;
 
+  &-filter-icons {
+    display: flex;
+    gap: 25px;
```
Comment (D1nviz) [inline code/src/pages/dashboard/dashboard-orders/DashboardOrdersPage.scss]: for gap we are using `spacing` from design-system

File: code/src/components/app-search-input/AppSearchInput.test.tsx (hunk 1 above)
Comment (OliaHrytsak) [reply code/src/components/app-search-input/AppSearchInput.test.tsx]: Fixed

File: code/src/components/app-search-input/AppSearchInput.tsx (hunk 2 above)
Comment (OliaHrytsak) [reply code/src/components/app-search-input/AppSearchInput.tsx]: Fixed

File: code/src/components/app-search-input/AppSearchInput.tsx (hunk 3 above)
Comment (OliaHrytsak) [reply code/src/components/app-search-input/AppSearchInput.tsx]: Fixed

File: code/src/components/app-search-input/AppSearchInput.tsx (hunk 4 above)
Comment (OliaHrytsak) [reply code/src/components/app-search-input/AppSearchInput.tsx]: Fixed

File: code/src/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.constants.ts (hunk 5 above)
Comment (OliaHrytsak) [reply code/src/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.constants.ts:4]: It looks like formatting already used)
Maybe too long path....

File: code/src/pages/dashboard/dashboard-orders/DashboardOrdersPage.tsx (hunk 7 above)
Comment (OliaHrytsak) [reply code/src/pages/dashboard/dashboard-orders/DashboardOrdersPage.tsx]: Fixed! Thank you for this advices!

File: code/src/pages/dashboard/dashboard-orders/messages/uk.json (hunk 8 above)
Comment (OliaHrytsak) [reply code/src/pages/dashboard/dashboard-orders/messages/uk.json]: Fixed

File: code/src/pages/dashboard/dashboard-orders/DashboardOrdersPage.scss (hunk 9 above)
Comment (OliaHrytsak) [reply code/src/pages/dashboard/dashboard-orders/DashboardOrdersPage.scss]: Fixed!

File: code/src/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.ts (hunk 6 above)
Comment (OliaHrytsak) [reply code/src/containers/dashboard-orders-filter-drawer/hooks/use-filtered-admin-orders/useFilteredAdminOrders.ts:6]: formatting already used)

**Review by D1nviz — APPROVED** [review]

**Review by heshvab — APPROVED** [review]

**Review by mhevyk — APPROVED** [review]
Amazing work was done by you! Now our order page for the administrator is filled with various useful functionality, which is good news! 😌

**Pull request merged**